### PR TITLE
Feat/ga/backfill unit tests

### PR DESCRIPTION
**Pull request by unseen1980**
# Issue

https://github.com/SAP/guided-answers-extension/issues/141

## Description

Backfilling UI tests

Before:
Screenshot 2022-07-08 at 14 03 24

After:
Screenshot 2022-07-08 at 14 02 47

## Checklist for Pull Requests

- [ ] Supplied as many details as possible on this change
- [ ] Included the link to the associated issue in the Issue section above
- [ ] The code conforms to the general [development guidelines](https://github.com/SAP/guided-answers-extension/blob/main/docs/developer-guide.md).
- [ ] The code has **unit tests** where applicable and is easily unit-testable
- [ ] This branch is appropriately named for the associated issue

### CODE REVIEW

**Review by Klaus-Keller — COMMENTED** [review]
Thanks @unseen1980 for adding the unit tests. Please see my comment below.

File: packages/webapp/src/webview/ui/components/GuidedAnswerNode/GuidedAnswerNode.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ function replace(domNode: DOMNode): ReactElement | undefined {
  * @param htmlString - HTML string
  * @returns - react elements including enhancements
  */
-function enhanceBodyHtml(htmlString: string): ReactElement | undefined {
+export function enhanceBodyHtml(htmlString: string): ReactElement | undefined {
```
[the remark this answers is not in the document]
Comment (unseen1980) [reply packages/webapp/src/webview/ui/components/GuidedAnswerNode/GuidedAnswerNode.tsx:55]: Yes only for the unit test. I don't know any other way. Any ideas to do it differently?

**Comment by Klaus-Keller** [comment]
@unseen1980, please check commit https://github.com/SAP/guided-answers-extension/pull/171/commits/1949a66ab533de6f553890793a836165af4181c1

The `enhanceBodyHtml()` and `replace()` functions are executed, if the `BODY` contains a `<span>`-tag with attribute [`HTML_ENHANCEMENT_DATA_ATTR_MARKER `](https://github.com/SAP/guided-answers-extension/blob/main/packages/types/src/types.ts#L74). This tag is added in function [`enhanceNode()`](https://github.com/SAP/guided-answers-extension/blob/main/packages/core/src/guided-answers-api.ts#L118) of module `@sap/guided-answers-extension-core` if the text to be replaced is found. It serves as an indicator for the UI and contains the command that needs to be executed onClick.

**Review by Klaus-Keller — APPROVED** [review]
Test code looks good. Approved.

**Comment by unseen1980** [comment]
Thanks @Klaus-Keller

**Pull request merged**